### PR TITLE
JAS_G_1: Introduces Database storage & Enhance the gateway routing.

### PR DESCRIPTION
**Pull request by MounicaSagi**
* Introduces Postgres DB & defines schema for consent_manager, bridge & bridge_service.

* Reads routing Information from cache.

* Hits DB when there is a cache miss.

* Adds tests.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -4,10 +4,13 @@ services:
...
       - "9002:8000"
     environment:
       SERVICE_TIMEOUT: 5
-      REGISTRY_PATH: /registry.yaml
+
+
+
+
+
+
```
Comment (ganesanarun) [inline docker-compose.yml]: remove this.

File: src/main/java/in/projecteka/gateway/GatewayConfiguration.java [hunk 2]
```diff
@@ -99,20 +102,46 @@ public String load(String key) {
...
+                          return "";
+                    }
+                });
+    }
+
+    @Bean({"bridgeMappings"})
+    public CacheAdapter<Pair<String, ServiceType>, String> createLoadingCacheAdapterForBridgeMappings() {
+        return new LoadingCacheAdapter(createMappingCacheForBridge(12));
+    }
+    public LoadingCache<Pair<String,ServiceType>, String> createMappingCacheForBridge(int duration) {
```
Comment (ganesanarun) [inline src/main/java/in/projecteka/gateway/GatewayConfiguration.java]: space between this two: nic

File: src/main/java/in/projecteka/gateway/common/MappingRepository.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+    private final PgPool dbClient;
+
+    public Mono<String> cmHost(String cmId) {
+        return Mono.create(monoSink -> dbClient.preparedQuery(SELECT_CM_MAPPING)
+                .execute(Tuple.of(cmId, true, false),
+                        handler -> {
+                            if (handler.failed()) {
+                                logger.error(handler.cause().getMessage(), handler.cause());
+                                monoSink.error(new DbOperationError());
```
Comment (ganesanarun) [inline src/main/java/in/projecteka/gateway/common/MappingRepository.java]: How valid `"Failed to persist in database"` is valid for this scenario?

**Comment by ganesanarun** [comment]
You removed the default mappings, how will things work when I run my gateway?

Can you put shell script/something which will connect to a DB and add default entries automatically which can be run after migration? which you can point out in the dev-setup.md too

**Review by ganesanarun — APPROVED** [review]

File: docker-compose.yml (hunk 1 above)
Comment (MounicaSagi) [reply docker-compose.yml]: done

File: src/main/java/in/projecteka/gateway/GatewayConfiguration.java (hunk 2 above)
Comment (MounicaSagi) [reply src/main/java/in/projecteka/gateway/GatewayConfiguration.java]: done

File: src/main/java/in/projecteka/gateway/common/MappingRepository.java (hunk 3 above)
Comment (MounicaSagi) [reply src/main/java/in/projecteka/gateway/common/MappingRepository.java]: modified to throw appropriate error msg

**Pull request merged**